go/database/mpt/live_trie.go

```diff
@@ -71,7 +71,7 @@ func VerifyFileLiveTrie(directory string, config MptConfig, observer Verificatio
 	if !exists {
 		return nil
 	}
-	return VerifyFileForest(directory, config, []Root{{
+	return VerifyCodesAndForest(directory, config, []Root{{
```

Comment:
I still think that this function should not check codes since LiveTrie instances are not managing codes, just the MPT structure. Codes are managed by `MptState` instances, and there should be a validation for those.

However, I guess there have been some discussions on this topic and this is the solution you got to. It will work, since right now there is no MPT without a Code store. We just need to be aware that if we ever have such a case, the verification will fail.

Reply:
I was proposing that each method `VerifyFileLiveTrie` and `VerifyArchive` should call first a new method that verifies the codes (could be called `VerifyMptState`?) and then the original `VerifyFileForest`.  It would decompose these two verifications into two independent steps.

Reply:
I am confused about it, because `VerifyFileLiveTrie` verifies the LiveDB (perhaps the method should be better named `VerifyLiveDB`) and it does contain codes, i.e. it should be verified as well.  It is like one instance of `MptState` in the case of live db while for the archive there is conceptually many `MptState` instances. On the other hand there is only one codes file in both cases, i.e. in both cases verify one instance of `MptState`.

Reply:
This discussion is a consequence of the missing cleanup of the dependency structure inside the MPT directory, which is pending since months. So ultimately, there is no clean solution.

Right now we have the following situation:
- a [`Forest`](https://github.com/Fantom-foundation/Carmen/blob/ac980c2cb1983277f771c061909164f9ae3b3a74/go/database/mpt/forest.go#L78) instance owns all MPT nodes, but not the codes
- a [`LiveTrie`](https://github.com/Fantom-foundation/Carmen/blob/ac980c2cb1983277f771c061909164f9ae3b3a74/go/database/mpt/live_trie.go#L29) is a wrapper around a Forest, so it also just owns the MPT nodes, no codes
- a [`MptState`](https://github.com/Fantom-foundation/Carmen/blob/ac980c2cb1983277f771c061909164f9ae3b3a74/go/database/mpt/state.go#L135) instance owns a Forest + the codes
- a [`ArchiveTrie`](https://github.com/Fantom-foundation/Carmen/blob/ac980c2cb1983277f771c061909164f9ae3b3a74/go/database/mpt/archive_trie.go#L32) owns a [`LiveState`](https://github.com/Fantom-foundation/Carmen/blob/ac980c2cb1983277f771c061909164f9ae3b3a74/go/database/mpt/state.go#L103) which is `MptState` in disguised, not at all what you would expect from the `Trie` in the `ArchiveTrie` name

Ideally, the package would be refactored such that everything that has a `Trie` in its name should only be concerned with MPT nodes, no codes. But this is not the case right now.

Following the ideal case, I would have the following expectations:
- a function called `verifyLiveTrie` checks the MPT nodes of a LiveDB
- a function called `verifyArchiveTrie` checks the MPT nodes of an Archive
- a function called `verifyMptState` checks the MPT nodes of the forest it owns + the codes

Right now, we have a
- `verifyCodesAndForest`
- `verifyArchive`
which basically circumvents the problem of the pending refactoring by adding new terms, that are not bound to existing types. `Archive` is neither an `ArchiveTrie` nor a `MptState`, but the way they are used together; `CodesAndForest` could be either a `LiveTrie` + `Code` file or a `ArchiveTrie` + `Code` file, but it assumes that the code file is named as the `MptState` is doing it; 

So my problem is: instead of improving the situation, or at least align to the existing setup, this naming adds additional confusion.

Reply:
I hope I understood your comment correctly. Could you check last commit? @HerbertJordan